src/sst/core/testingframework/test_engine_support.py

```diff
@@ -0,0 +1,326 @@
+    """
+###
+
+    def __init__(self, cmd_str, output_file_path=None, error_file_path=None,
+                 set_cwd=None, use_shell=False):
+        self._output_file_path = None
+        self._error_file_path = None
+        self._cmd_str = None
+        self._process = None
+        self._timeout_sec = 60
+        self._run_status = None
+        self._run_output = ''
+        self._run_error = ''
+        self._run_timeout = False
+        self._use_shell = use_shell
+        self._set_cwd = set_cwd
+        self._validate_cmd_str(cmd_str)
+        self._output_file_path = self._validate_output_path(output_file_path)
+        self._error_file_path = self._validate_output_path(error_file_path)
+
+####
+
+    def run(self, timeout_sec=60, **kwargs):
+        """ Run a command then return and OSCmdRtn object. """
+        if not (isinstance(timeout_sec, (int, float)) and not isinstance(timeout_sec, bool)):
+            raise ValueError("ERROR: Timeout must be an int or a float")
+
+        self._timeout_sec = timeout_sec
+
+        # Build the thread that will monitor the subprocess with a timeout
+        thread = threading.Thread(target=self._run_cmd_in_subprocess, kwargs=kwargs)
+        thread.start()
+        thread.join(self._timeout_sec)
+        if thread.is_alive():
+            self._run_timeout = True
+            self._process.terminate()
+            thread.join()
+
+        # Build a OSCommandResult object to hold the results
+        rtn = OSCommandResult(self._cmd_str, self._run_status, self._run_output,
+                              self._run_error, self._run_timeout)
+        return rtn
+
+####
+
+    def _run_cmd_in_subprocess(self, **kwargs):
+        """ Run the command in a subprocess """
+        file_out = None
+        file_err = None
+        saved_cwd = os.getcwd()
+
+        try:
+            cwd = os.getcwd()
+
+            if self._set_cwd is not None:
+                os.chdir(os.path.abspath(self._set_cwd))
+
+            # If No output files defined, default stdout and stderr to normal output
+            if 'stdout' not in kwargs and self._output_file_path is None:
+                kwargs['stdout'] = subprocess.PIPE
+            if 'stderr' not in kwargs and self._error_file_path is None:
+                kwargs['stderr'] = subprocess.PIPE
+
+            # Create the stderr & stdout to the output files, if stderr path is
+            # not defined, then use the normal output file
+            if 'stdout' not in kwargs and self._output_file_path is not None:
+                file_out = open(self._output_file_path, 'w+')
+                kwargs['stdout'] = file_out
+                if self._error_file_path is None:
+                    kwargs['stderr'] = file_out
+            if 'stderr' not in kwargs and self._error_file_path is not None:
+                file_err = open(self._error_file_path, 'w+')
+                kwargs['stderr'] = file_err
+
+            self._process = subprocess.Popen(self._cmd_str,
```

Comment:
We could... but the class supports the use_shell option.  If use_shell is true, the command must come it as a cmd_str otherwise the items in the list after the first item are sent as parameters to the shell.  We could make this work, but its a lot of changes across the board on code that has been working solidly for a while.  I suggest we write it up as an issue against the frameworks.